### PR TITLE
openInNewTab

### PR DESCRIPTION
**Pull request by magdyheiba**

### CODE REVIEW

File: src/main/java/com/shaft/gui/element/ElementActions.java [hunk 1]
```diff
@@ -533,7 +533,15 @@ public static void switchToIframe(WebDriver driver, By elementLocator) {
...
         WebDriverElementActions.switchToWindow(driver, nameOrHandle);
     }
-
+    /**
+     * Switches focus to another Tap
+     *
+     * @param driver       the current instance of Selenium webdriver
+     * @param URL The name of the URL you want to navigate to
+     */
+    public static void switchToNewTap(WebDriver driver, String URL) {
```
Comment (MohabMohie) [inline src/main/java/com/shaft/gui/element/ElementActions.java]: opening a link in a new tab is not an element action.
kindly move it to browser actions instead.
also please note that every method has two variants, one is static and one is a regular method. the static method takes a driver object as a parameter and the regular method uses the existing driver object.

File: src/main/java/com/shaft/gui/element/WebDriverElementActions.java [hunk 2]
```diff
@@ -1146,6 +1147,26 @@ public static void switchToWindow(WebDriver driver, String nameOrHandle) {
             WebDriverElementActions.failAction(driver, nameOrHandle, null);
         }
     }
+    /**
+     * Switches focus to another Tap
+     *
+     * @param driver       the current instance of Selenium webdriver
+     * @param URL The name of the URL you want to navigate to
+     */
+    public static void switchToNewTap(WebDriver driver, String URL) {
```
Comment (MohabMohie) [inline src/main/java/com/shaft/gui/element/WebDriverElementActions.java]: the method should validate that the link has been opened in a new tab successfully.
this required getting window handles before and after opening the new tab, and confirming that the new tab was opened, printing the new tab window handle in the console using logdiscrete

**Review by MohabMohie — CHANGES_REQUESTED** [review]
thank you for your valuable contribution. kindly check the requested comments.

File: src/main/java/com/shaft/gui/element/WebDriverElementActions.java (hunk 2 above)
Comment (magdyheiba) [reply src/main/java/com/shaft/gui/element/WebDriverElementActions.java]: Done(Y)

File: src/main/java/com/shaft/gui/element/ElementActions.java (hunk 1 above)
Comment (magdyheiba) [reply src/main/java/com/shaft/gui/element/ElementActions.java]: Done (Y)

**Review by MohabMohie — APPROVED** [review]
Thanks for your valued contribution!

**Pull request merged**